homework/sheet02.md

```diff
@@ -2,86 +2,70 @@
 archetype: assignment
 title: "Blatt 02: CFG"
 author: "BC George, Carsten Gips (HSBI)"
+points: "10 Punkte"
 
 hidden: true
 
 sketch: true
 ---
 
+<!--  pandoc -s -f markdown -t markdown+smart-grid_tables-multiline_tables-simple_tables --columns=94 --reference-links=true  sheet02.md  -o xxx.md  -->
 
-## A2.1: Grammatik für Mini-Python
+## A2.1: PDA (2P)
 
-Erstellen Sie eine Grammatik für [**Mini-Python**].
+Erstellen Sie eine deterministischen PDA, der die Sprache
+$L = \{w \in {a, b, c}^* | w\ \text{hat doppelt so viele a's wie c's}\}$ akzeptiert.
 
-Der in der [Dokumentation] definierte [syntaktische] und [semantische]
-Sprachumfang soll mit Ihrer Grammatik mindestens unterstützt werden.
-Dabei ist die Funktionalität wie in Python mit folgenden Ausnahmen:
+Beschreiben Sie Schritt für Schritt, wie der PDA die Eingaben *bcaba* und *bccac* abarbeitet.
 
-*   Einrückung ist für die Funktionalität irrelevant
-*   Schleifen, Funktionen und Klassen werden mit `#end` beendet
 
-Nachfolgend einige Beispiele in Ergänzung zur [Dokumentation]:
+## A2.2: Akzeptierte Sprache (1P)
 
-1.  Beispiele für IF-Statements:
+Ist der folgenden PDA deterministisch? Warum (nicht)?
 
-    ```python
-    a= 2
-    if a == 2:
-        print("a is ", a)
-    #end
-    ```
 
-    ```python
-    if a == 2:
-    print("a is ", a)
-    elif a == 3:
-    print("a is ", a)
-    else:
-    print("a is neither 2 and 3")
-    #end
-    ```
+Bild folgt
 
-2.  Beispiel für Schleifen:
+Welche Sprache akzeptiert er?
 
-    ```python
-    while(x>y):
-        print(x ,"is bigger than ",y)
-        x=x-1
-    #end
-    ```
 
-3.  Beispiel für Funktionen:
+## A2.3: Epsilon-Übergänge (2P)
 
-    ```python
-    def foo():
-    print("ich bin eine Funktion")
-    #end
-    ```
+Entwickeln Sie einen Algorithmus, der einen PDA mit $\epsilon$ - Übergängen in einen ohne diese umwandelt.
 
-4.  Beispiele für Klassen:
 
-    ```python
-    class A:
-    def foo(self):
-        print("Ich bin eine Methode von A")
-    #end
-    #end
-    ```
+## A2.4 Kontextfreie Sprache (1P)
 
-    ```python
-    class B(A): #Vererbung B erbt von A
-    def foo(self):
-        print("Ich bin eine Methode von B")
-    #end
-    #end
-    ```
+Welche Sprache generiert die folgende kontextfreie (Teil-) Grammatik?
+
+$G = (\{Statement, Condition, \ldots\}, \{"if", "else", \ldots\}, P, Statement)$
+
+mit $P =$
+
+$\{Statement \rightarrow "if"\ Condition\ Statement\ | \ "if"\ Condition\ Statement\ "else"\ Statement$
+
+$Condition \rightarrow \ldots\}$
```

Comment:
```suggestion
$Condition \rightarrow \ldots \rbrace$
```

Reply:
Hmmm. Damit MathJax mit den geschweiften Klammern umgehen kann, **müssen** die in der selben Mathe-Umgebung sein. 

<img width="1012" alt="Screenshot 2024-10-22 at 08 27 56" src="https://github.com/user-attachments/assets/10d905d8-1fd9-48e2-9293-c71d9c861e6d">

Reply:
Im Skript ist der zweite Teil mit der `$Condition \rightarrow \ldots\}$` sogar komplett weg. Das sollte wirklich alles in die selbe Mathe-Umgebung.